### ISSUE TITLE
New checker | create instance attributes in __init__ only

### ISSUE THREAD

**Issue by sushobhit27** [PyCQA/pylint#2398]
### Is your feature request related to a problem? Please describe
```
class Fake:
    def __init__(a, b):
        self.attr1 = a
        self.attr2 = b

    def meth1(self, c):
        self.attr3 = c
```

### Describe the solution you'd like
There should be warning that attr3 is only available for object on which meth1 is called, instead on all instances of Fake class or like "create instance attributes in __init__ only".

### Additional context
Add any other context about the feature request here.

**Comment by PCManticore** [comment]
We already have `attribute-defined-outside-init` for this

**Comment by sushobhit27** [comment]
@PCManticore ok I didn't know as it is not raised in above code sample.

$ pylint sample1.py
************* Module sample1
sample1.py:2:4: E0213: Method should have "self" as first argument (no-self-argument)
sample1.py:3:8: E0602: Undefined variable 'self' (undefined-variable)
sample1.py:4:8: E0602: Undefined variable 'self' (undefined-variable)

----------------------------------------------------------------------
Your code has been rated at -15.00/10 (previous run: 10.00/10, -25.00)

**Comment by PCManticore** [comment]
Mm, interesting, not sure why it's not getting emitted though.

**Comment by bart2** [comment]
FYI, it is possible that OP has `.pylintrc` present, with `disable=R,C,W`, to report only errors. The `Attribute '...' defined outside __init__` is a warning level check. To force all levels to be reported on the command line, you can use:
```
pylint -e R,C,W,E sample1.py
```
